src/ECharts/Event.js

```diff
@@ -1,11 +1,123 @@
+                if (option[axisKey][j].gridIndex === gridIndex) {
+                    addData(e.areas[i], option[axisKey][j], setKey);
+                    e.areas[i][axisIndexKey] = j;
+                }
+            }
+        }
+    }
+    return e;
+}
+
+function addXAxisData(option, e) {
+    return addAxisData(option, e, "xAxis", "xAxisData", "xAxisIndex");
+}
+
+function addYAxisData(option, e) {
+    return addAxisData(option, e, "yAxis", "yAxisData", "yAxisIndex");
+}
+
+function addSeriesAndTitles(option, e) {
+    var i, xIx, yIx, serie, ai, area;
+    for (ai = 0; ai < e.areas.length; ai++) {
+        area = e.areas[ai];
+        area.serieNames = [];
+        area.serieIndices = [];
+        for (i = 0; i < option.series.length; i++) {
+            serie = option.series[i];
+            xIx = serie.xAxisIndex || 0;
+            yIx = serie.xAxisIndex || 0;
+            if (xIx == area.xAxisIndex && yIx == area.yAxisIndex) {
+                area.serieNames.push(serie.name);
+                area.serieIndices.push(i);
+            }
+        }
+    }
+    var titles, titleTexts = [];
+    if (!option.title) {
+        titles = [];
+    } else if (typeof option.title.length === "undefined") {
+        titles = [option.title];
+    } else {
+        titles = option.title;
+    }
+    for (i = 0; i < titles.length; i++) {
+        if (titles[i].text) titleTexts.push(titles[i].text);
+    }
+    e.titleTexts = titleTexts;
+    return e;
+}
+
+function addAll(option, e) {
+    return addSeriesAndTitles(option, addYAxisData(option, addXAxisData(option, e)));
+}
+
 exports.on_ = function(chart) {
     return function(eName) {
         return function(callback) {
             return function() {
                 return chart.on(eName, function(e) {
-                    callback(e)();
-                });
+                    var modifiedE = eName === "brush" ? addAll(this.getOption(), e) : e;
+                    callback(modifiedE)();
+                    });
             };
         };
     };
 };
+
+exports.dispatchAction_ = function(maybeKeys) {
+    return function(maybe) {
+        return function(rawAction) {
+            return function(chart) {
+                return function() {
+                    var keys = Object.keys(rawAction),
+                        i, key, value, keysArr,
+                        action = {};
+                    for (i = 0; i < keys.length; i++) {
+                        key = keys[i];
+                        if (typeof maybeKeys[rawAction.type] === "undefined") {
+                            keysArr = [];
+                        } else {
+                            keysArr = maybeKeys[rawAction.type];
+                        }
+                        if (keysArr.indexOf(key) == -1) {
+                            value = rawAction[key];
+                        } else {
+                            value = maybe(undefined)(function(x) {return x;})(rawAction[key]);

```

Comment:
We can pass `fromMaybe` instead of `maybe` (based on it's usage).
`rawAction` is normal js object right? so why do we need `maybe` for rawAction[key]? Also `maybe` is used only here not in upper branch, why?

Reply:
`rawAction` is purescript object :) I'll switch to `fromMaybe`

Reply:
Ah so `maybeKeys`  is from type to array of such keys which contain js values wrapped in maybe? in various `EChartsEvents` I couldn't see any `Maybe` field

Reply:
like `dataIndex` is just `Int` so, where the Maybe comes from